src/main/scala/org/caoilte/spray/routing/LogAccessRouting.scala

```diff
@@ -55,7 +56,27 @@ class RequestAccessLogger(ctx: RequestContext, accessLogger: AccessLogger,
       accessLogger.logAccess(request, response, timeStampCalculator(Unit))
       forwardMsgAndStop(response)
     }
+    case Confirmed(ChunkedResponseStart(response), sentAck) => {
+      // In case a chunk transfer is started, store the response for later logging
+      chunkedResponse = response
+      responder forward Confirmed(ChunkedResponseStart(response), sentAck)
+    }
+    case Confirmed(MessageChunk(data, extension), sentAck) => {
+      // Each junk is considered a heart-beat: We are still producing output and the client is still consuming
```

Comment:
might be able to re-write this

case confirmed@Confirmed(mc:MessageChunk,sentAck) => {
    //restart timeout thread
   responder forward confirmed
}